src/GraphQL/Types/Scalars/StringGraphType.cs

```diff
@@ -16,5 +16,8 @@ public class StringGraphType : ScalarGraphType
 
         /// <inheritdoc/>
         public override bool CanParseLiteral(IValue value) => value is StringValue;
+
+        /// <inheritdoc/>
+        public override IValue ToAST(object value) => new StringValue(value.ToString());
```

Comment:
NRE if value is `null`. Also what if `value` is of "wrong" type? For example, `FileStream` or `Image` ?

Reply:
> NRE if value is null

Not applicable. We can change it if you want.  It really should throw an `ArgumentNullException` as that is invalid for the method.  Seems like a waste of time.

> Also what if value is of "wrong" type? 

What is a "wrong" type? I would imagine that people might pass back other objects expecting them to be serialized by `System.Text.Json` to a string, such as date/time types or a Uri.  We can't be sure, so better safe than sorry. This won't be the same as a Json serializer will do but better than nothing,

Reply:
> Not applicable

`new StringGraphType().ToAST(null)` -> NRE

> What is a "wrong" type?

I don't know how I should exactly name it. I want to say that each scalar has a corresponding .NET type known to it (or several):
`UriGraphType` - `Uri`
`FloatGraphType` - `float`, `double`, `int`, `long` (many)
`DateTimeGraphType` - `DateTime`, `string`

But what about `StringGraphType`. Of course it has a corresponding `string` .NET type. But what **else** ? Now I see that:
1. `ParseLiteral` handles only StringValue
2. `ParseValue` handles any value - is it right?
3. `CanParseLiteral` handles only StringValue
4. `ToAST` handles ... any value ?   Is it right?

Isn't it worth limiting such an "omnivorous" of `StringGraphType`? Maybe not, I'm not sure. Perhaps this behavior is flexible, but it may also hide errors.

Reply:
I'm okay to change this for v4.  Should not be changed during an interim version.  Will break code where people incorrectly return a number instead of a string, expecting GraphQL to cast it to a string automatically.  If you think we should do this also, let's do it in a separate PR.

Reply:
I'm still not sure. I don't know which option is better.

Reply:
#2335